qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,319 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Converter to convert a problem with equality constraints to unconstrained with penalty terms."""
+
+import copy
+import itertools
+import logging
+from math import fsum
+from typing import Optional, cast, Union, Tuple, List
+
+import numpy as np
+
+from .quadratic_program_converter import QuadraticProgramConverter
+from ..exceptions import QiskitOptimizationError
+from ..problems.constraint import Constraint, ConstraintSense
+from ..problems.quadratic_objective import QuadraticObjective
+from ..problems.quadratic_program import QuadraticProgram
+from ..problems.variable import Variable
+
+logger = logging.getLogger(__name__)
+
+
+class LinearInequalityToPenalty(QuadraticProgramConverter):
+    r"""Convert a problem of known constraints to unconstrained with penalty terms.
+
+    There are known constraints which do not require to add slack variables to
+    construct penalty terms [1].
+
+    Supported known constraint in this class is shown below.
+
+    .. math::
+      \begin{array}{|l|l|}
+      \hline \text { Classical Constraint } & \text { Equivalent Penalty } \\
+      \hline x+y \leq 1 & P(x y) \\
+      \hline x+y=1 & P(1-x-y+2 x y) \\
+      \hline x \leq y & P(x-x y) \\
+      \hline x_{1}+x_{2}+x_{3} \leq 1 & P\left(x_{1} x_{2}+x_{1} x_{3}+x_{2} x_{3}\right) \\
```

Comment:
For simple cases, `x_1`, `x_2`, `x_3` are enough.

Reply:
You can extend this case for x_1,...,x_n, i.e., `\sum_i^n x_i <= 1` => `P (\sum_{i, j : i < j} x_i x_j)`

Reply:
For the extension, please refer to y >= x (GE) case below.

Reply:
GE case is difficult to extend. You just need to generalize LE case now.

Reply:
See https://github.com/Qiskit/qiskit-optimization/pull/157#issuecomment-858221650